rezerwacje-front/src/views/Exams/components/ExamsContent/ExamsContent.jsx

```diff
@@ -33,11 +36,26 @@ const ExamsContent = () => {
         });
     };
 
-    const recruitmentSpaces = ["lato 2020", "zima 2021"];
+    useEffect(() => {
+        fetch("/api/recruitment/all")
+            .then(res => res.json())
+            .then(json => {
+                setRecruitments(json["recruitments"]);
+            })
+            .catch(e => console.log(e));
+    }, []);
 
-    const recruitmentPeriods = ['1', '2'];
+    useEffect(() => {
+        if (values.recruitment.length > 0)
+            fetch(`/api/recruitment-period/recruitment/${values.recruitment}`)
+                .then(res => res.json())
```

Comment:
TBH I had some serious trouble here, it didn't work. `values.recruitment` can't be a string.

The problem is, `/api/recruitment-period/recruitment/${values.recruitment}` endpoint expects recruitment ID (integer value), not the name. But you can't use ID either, because in the Add Exam dialog `/api/recruitment/${props.recruitment}` endpoint is used, which expects the name (string value) of recruitment. 

So the backend is messed up, to work with it you need to keep both ID and name of recruitment in `values` variable. Best option would be to fix the backend to have uniform API

Reply:
NIAK-91 should resolve that but still ID must be used in requests

Reply:
Sorry, I forgot to add that I used backend from NIAK-91